internal/schemautil/schemautil.go

```diff
@@ -161,7 +161,7 @@ func EmptyObjectDiffSuppressFuncSkipArrays(s map[string]*schema.Schema) schema.S
 	return func(k, old, new string, d *schema.ResourceData) bool {
 		for _, key := range skipKeys {
 			if strings.Contains(k, fmt.Sprintf(".%s.", key)) {
-				return false
+				return !d.HasChange(k)
```

Comment:
could this be implemented by applying a different approach? or perhaps postponed? I believe with the release of v4.0.0, this should be resolved automatically when we implement the use of `null` types